### ISSUE TITLE
Is training your waveRNN on piano music realistic ?

### ISSUE THREAD

**Issue by VincentMatter** [fatchord/WaveRNN#81]
Hello,

I'm a master student doing a thesis on music generation. My goal is to condition a neural network on emotion so as to generate "sad piano music" or "happy piano music" or "angry piano music", you got me. I know WaveRNN has been mostly used for speech generation. Is my goal achievable with your implementation ? If so, how long would the training take and how much data would I need ?

Thank you for your help

**Comment by fatchord** [comment]
@VincentMatter While I can't say for certain (since I haven't tried it), I would say that it's very likely to succeed to some degree. 

I would offer the following advice. Since you don't need the mel spectrogram conditioning network, I would dump that for now. All you need for the input are vectors representing happy, sad etc and the previously generated samples. I would start off using an 8bit softmax model since it is much much faster to train - also start at a lower samplerate like 16kHz. As for dataset size - I'd start small - one track per emotion, see if that works and then build on top of that.

**Comment by VincentMatter** [comment]
Thanks for the quick reply and the info, I will make sure to heed your advice. How long would it take approximately  for the initial experiment (one track per emotion, 16khz and 8bit softmax model) on a GeForce GTX 960M ?
I just need to know if it's worth using my GPU or if I should head toward cloud computing.

Once I'm done, I'll make sure to send you the results

**Comment by fatchord** [comment]
@VincentMatter I've no idea how that GPU will perform. I guess the best thing to try is download the repo and see what steps/s you get just training wavernn on a normal dataset. For comparison, I get around 3steps/s on a GTX1080

**Comment by TheButlah** [comment]
A GTX 960M is a mobile GPU, and not an especially powerful one. I would advise that you use cloud computing, because you will probably have to train many times to iterate on your experiments. Alternatively, just use a regular desktop class (no M suffix) GPU like a 1080 or better

**Comment by gauripravishi** [comment]
What was the result finally, I am a master's student doing the same research (almost) Were you able to generate music using WaveRNN?
